hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/NearCachedClientCacheProxy.java

```diff
@@ -279,24 +289,22 @@ private void populateResultFromNearCache(Collection<?> keys, List<Object> result
         }
     }
 
-    private Map<Object, Long> getNearCacheReservations(Collection<?> nearCacheKeys, Map<Object, Data> keyMap) {
-        Map<Object, Long> reservations = createHashMap(nearCacheKeys.size());
-        for (Object key : nearCacheKeys) {
-            Data keyData = serializeKeys ? (Data) key : keyMap.get(key);
-            long reservationId = tryReserveForUpdate(key, keyData);
+    private void createNearCacheReservations(Collection<?> dataKeys, Map<Object, Long> reservations) {
```

Comment:
How about this for clarity (since you cast to `Data` anyway) : 
```
    private void createNearCacheReservations(Collection<Data> dataKeys, Map<Object, Long> reservations) {
        for (Data key : dataKeys) {
            long reservationId = tryReserveForUpdate(key, key);
            if (reservationId != NOT_RESERVED) {
                reservations.put(key, reservationId);
            }
        }
    }
```
And when calling cast the collection : `createNearCacheReservations((Collection<Data>) ncKeys, reservations);`